app/services/mixtape_generator.rb

```diff
@@ -0,0 +1,20 @@
+class MixtapeGenerator
+  def initialize(genre:)
+    @genre = genre
+  end
+
+  def self.perform(genre:)
+    new(genre: genre).perform
+  end
+
+  def perform
+    Mixtape.new(genre: @genre, songs: pick_songs)
+  end
+
+  def pick_songs
+    Song.order_by_rand
+        .includes(:genre, :artist)
+        .joins(:genre, :artist)
+        .where("lower(genres.name) = ?", @genre.downcase)
```

Comment:
That makes sense...will change to ```genre_name```. I hadn't taken a step back to look at it but you are right that it is misleading since there is a ```Genre``` class, but also ```params[:genre]``` which corresponds to ```genre.name```.